client/components/pages/admin/WorkSpaceUser.vue

```diff
@@ -0,0 +1,306 @@
+<template>
+    <div>
+        <h4 class="font-bold">Members</h4>
+        <form v-if="isWorkspaceAdmin" @submit.prevent="addUser" class="my-2">
+          <text-input
+            name="email"
+            v-model="newUser"
+            label="Email"
+            :required="true"
+            placeholder="Add a new user by email"
+          />
+          <select-input
+            name="newUserRole"
+            v-model="newUserRole"
+            :options="roleOptions"
+            placeholder="Select User Role"
+            label="Role"
+            :required="true"
+          />
+          <div class="flex justify-end mt-2">
+            <v-button
+            color="outline-blue"
+            :loading="loadingUsers"
+          >
+            Add User
+          </v-button>
+          </div>
+        </form>
+
+        <UTable
+          :loading="loadingUsers"
+          :rows="rows"
+          :columns="columns"
+        >
+        <template #actions-data="{ row, index }" v-if="isWorkspaceAdmin">
+          <UButton
+            @click="editUser(index)"
+            icon="i-heroicons-pencil"
+            size="2xs"
+            color="blue"
+            variant="outline"
+            :ui="{ rounded: 'rounded-full' }"
+            square
+          />
+
+          <UButton
+            @click="removeUser(index)"
+            icon="i-heroicons-trash"
+            size="2xs"
+            color="red"
+            variant="outline"
+            :ui="{ rounded: 'rounded-full' }"
+            square
+          />
+        </template>
+        </UTable>
+
+        <modal
```

Comment:
Please create a separate component for this

Reply:
To keep the state simple, it can simply take as a prop a user, and emit an event on update so that data can be updated back in this component